source/common/common/thread.cc

```diff
@@ -13,6 +13,15 @@ bool MainThread::isMainThread() {
   return main_thread_singleton->inMainThread() || main_thread_singleton->inTestThread();
 }
 
+bool MainThread::isWorkerThread() {
+  auto main_thread_singleton = MainThreadSingleton::getExisting();
+  if (main_thread_singleton == nullptr) {
+    return true;
```

Comment:
add comment here that this would occur in a unit-test scenario, and in that scenario, where this is called, we can return 'true' here to allow worker-thread assertions to pass.

Reply:
Thanks, comment added.